code gen error on return if-unwrapping

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
OS: linux, Ubuntu 18.04.5 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i5-4200U CPU @ 1.60GHz
CC version: cc (Ubuntu 10.1.0-2ubuntu1~18.04) 10.1.0

getwd: /home/donp/projects/ircbr
vmodules: /home/donp/.vmodules
vroot: /opt/v/2021.6
vexe: /opt/v/2021.6/v
vexe mtime: 2021-03-17 17:40:01
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.2 98c611d.c2b5743

Git version: git version 2.29.0
Git vroot status: weekly.2021.11-44-gc2b57438 (17 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 7543de81

```v
struct Thing {
	name string
}

fn main() {
	option_check('zoo')
}

fn option_check(name string) string {
	return if thing := find_thing_by_name(name) { thing.name } else { 'safename' }
}

fn find_thing_by_name(name string) ?&Thing {
	return &Thing{
		name: 'zs'
	}
}
```
```c
$ v -cg run guard.v
/tmp/v/guard.10361233136010102390.tmp.c:8751: error: identifier expected

string main__option_check(string name) {
        string _t96 = (/* guard */ ? (thing->name) : (_SLIT("safename")));
        return _t96;
}
```

The first part of the ternary operator has no term.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4353,6 +4353,9 @@ fn (mut g Gen) need_tmp_var_in_if(node ast.IfExpr) bool {
 			return true
 		}
 		for branch in node.branches {
+			if branch.cond is ast.IfGuardExpr {
+				return true
+			}
 			if branch.stmts.len == 1 {
 				if branch.stmts[0] is ast.ExprStmt {
 					stmt := branch.stmts[0] as ast.ExprStmt

--- a/vlib/v/tests/if_guard_test.v
+++ b/vlib/v/tests/if_guard_test.v
@@ -59,3 +59,22 @@ fn test_chan_pop() {
 	assert res == [6.75, -3.25, -37.5]
 }
 
+struct Thing {
+	name string
+}
+
+fn test_return_if_guard() {
+	ret := option_check('zoo')
+	println(ret)
+	assert ret == 'zs'
+}
+
+fn option_check(name string) string {
+	return if thing := find_thing_by_name(name) { thing.name } else { 'safename' }
+}
+
+fn find_thing_by_name(name string) ?&Thing {
+	return &Thing{
+		name: 'zs'
+	}
+}

